tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+""" Utilities used to test the different subclasses of the ProfetTask. """
+from pathlib import Path
+from typing import ClassVar, Type
+import pytest
+
+from orion.benchmark.task.base import BaseTask
+from orion.benchmark.task.profet.profet_task import (
+    MetaModelTrainingConfig,
+    ProfetTask,
+    download_data,
+    load_data,
+)
+from orion.algo.space import _Discrete
+from typing import Dict, Tuple
+import numpy as np
+import torch
+
+from logging import getLogger as get_logger
+from .conftest import REAL_PROFET_DATA_DIR, requires_profet_data
+
+logger = get_logger(__name__)
+shapes: Dict[str, Tuple[Tuple[int, ...], Tuple[int, ...], Tuple[int, ...]]] = {
+    "fcnet": ((600, 6), (27, 600), (27, 600)),
+    "forrester": ((10, 1), (9, 10), (9, 10)),
+    "svm": ((200, 2), (26, 200), (26, 200)),
+    "xgboost": ((800, 8), (11, 800), (11, 800)),
+}
+y_min: Dict[str, float] = {
+    "fcnet": 0.0,
+    "forrester": -18.049155413936802,
+    "svm": 0.0,
+    "xgboost": 0.0,
+}
+y_max: Dict[str, float] = {
+    "fcnet": 1.0,
+    "forrester": 14718.31848526001,
+    "svm": 1.0,
+    "xgboost": 3991387.335843141,
+}
+c_min: Dict[str, float] = {
+    "fcnet": 0.0,
+    "forrester": -18.049155413936802,
+    "svm": 0.0,
+    "xgboost": 0.0,
+}
+c_max: Dict[str, float] = {
+    "fcnet": 14718.31848526001,
+    "forrester": 14718.31848526001,
+    "svm": 697154.4010462761,
+    "xgboost": 5485.541382551193,
+}
+
+
+@pytest.fixture(autouse=True)
+def load_fake_data(monkeypatch, tmp_path_factory):
+    """ Fixture that prevents attempts to download the true Profet datasets, and instead generates
+    random data with the same shape.
+    """
+
+    import orion.benchmark.task.profet.profet_task
+
+    real_load_data = orion.benchmark.task.profet.profet_task.load_data
+
+    def _load_data(path: Path, benchmark: str):
```

Comment:
That could be confusing. The actual test depends on the state of the file-system. I would be rather in favor of parametrizing this fixture, running one with fake data and one with real profet data if we plan to run tests with real profet data as well. By default tests on real profet data would be skipped as the data is not available, but at least it would be easy to run both locally if we have the data.

Reply:
Fixed: A fixture is parametrized to run tests with real data (skipped when not available) and fake data.

Reply:
Sounds good! I'd like to get a clarification on this docstring however: https://github.com/Epistimio/orion/pull/632/files#r770900915

Reply:
Sorry, the above-link seems broken. Is this still an issue? If so, what docstring needs to be updated?

Reply:
Oops indeed. I think that was referring to the docstring of `mock_load_data`, but the current version seems fine. I'll mark this as resolved.